server/analyzer/optimize_functions.go

```diff
@@ -31,12 +31,35 @@ func OptimizeFunctions(ctx *sql.Context, a *analyzer.Analyzer, node sql.Node, sc
 	if scope != nil && scope.CurrentNodeIsFromSubqueryExpression {
 		return node, transform.SameTree, nil
 	}
-	return pgtransform.NodeExprsWithNodeWithOpaque(node, func(node sql.Node, expr sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
+	hasSRF := false
+	funcName := ""
+	var function sql.Expression
+	var tempType sql.Type
+	node, same, err := pgtransform.NodeExprsWithNodeWithOpaque(node, func(node sql.Node, expr sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
 		if compiledFunction, ok := expr.(*framework.CompiledFunction); ok {
+			funcName = compiledFunction.Name
+			hasSRF = compiledFunction.IsSRF()
+			if hasSRF {
+				// TODO: need a way to get the return type from the function
+				//  this is temporary way to get the return type
+				tempType = compiledFunction.Arguments[0].Type()
+			}
 			if quickFunction := compiledFunction.GetQuickFunction(); quickFunction != nil {
+				function = quickFunction
 				return quickFunction, transform.NewTree, nil
+			} else {
+				function = compiledFunction
 			}
 		}
 		return expr, transform.SameTree, nil
 	})
+
+	if err != nil {
+		return nil, transform.NewTree, err
+	}
+	if hasSRF {
+		node = pgnode.NewSetReturningFunctionTable(funcName, function, tempType)
+	}
```

Comment:
I feel that this should be in the `pgtransform` function. If there were multiple SRFs as a node's child then this will only capture the last one seen, which doesn't seem correct.